File: T4/T4_ViewPager/app/src/main/java/com/system/develop/t4_viewpager/fragments/FragmentDos.java
package com.system.develop.t4_viewpager.fragments;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.system.develop.t4_viewpager.R;

public class FragmentDos extends Fragment{

    View v;
    TextView textView;

    public static FragmentDos newInstance(String asd) {
        
        Bundle args = new Bundle();
        
        FragmentDos fragment = new FragmentDos();
        fragment.setArguments(args);
        return fragment;
    }

    public void cambiarDatos(String s){
        textView.setText(s);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        v = inflater.inflate(R.layout.fracment_dos_layout,container,false);
        textView = v.findViewById(R.id.comunicacionf1);
        return v;
    }
}

File: Kotlin/RECUPERACION/Naviegacion/app/src/main/java/com/example/navegacion/FragmentDos.kt
package com.example.navegacion

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import com.example.navegacion.databinding.FragmentDosBinding
import com.example.navegacion.databinding.FragmentUnoBinding

class FragmentDos : Fragment() {

    private lateinit var binding: FragmentDosBinding
    private var nombre: String? = null;

    override fun onAttach(context: Context) {
        super.onAttach(context)
        nombre = arguments?.getString("nombre")
    }
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        binding = FragmentDosBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.botonF2.setOnClickListener {
           findNavController().navigate(R.id.action_fragmentDos_to_fragmentUno)
        }
    }

}